src/arch/dotproductneon.cpp

```diff
@@ -0,0 +1,57 @@
+#if !(defined(__arm__) || defined(__aarch64__))
+
+#include "dotproductneon.h"
+#include <cstdio>
+#include <cstdlib>
+
+namespace tesseract {
+    int32_t IntDotProductNEON(const int8_t* u, const int8_t* v, int n) {
+        fprintf(stderr, "IntDotProductNEON can't be used\n");
+        abort();
+    }
+}  // namespace tesseract
+
+#else  // !(defined(__arm__) || defined(__aarch64__))
+
+#include "dotproductneon.h"
```

Comment:
given that you are including this header in both the cases, you might want to move this include outside of #if-#else-#endif

Reply:
Done